test/runners.jl

```diff
@@ -128,6 +128,41 @@ end
                 @test endswith(readchomp(iobuff), "Hello World!")
             end
 
+            @testset "C++ - $(platform)" for platform in platforms
+                ur = preferred_runner()(dir; platform=platform)
+                iobuff = IOBuffer()
+                test_c = """
+                #include <iostream>
+                class breakCCompiler; // Courtesy of Meson
+                int main() {
+                    std::cout << "Hello World!" << std::endl;
+                    return 0;
+                }
+                """
+                test_script = """
+                set -e
+                # Install `CompilerSupportLibraries_jll` v0.5.0 in the `\${prefix}` to make
+                # sure it doesn't break compilation of the program
+                mkdir -p \${prefix}
+                wget -qO - "https://github.com/JuliaBinaryWrappers/CompilerSupportLibraries_jll.jl/releases/download/CompilerSupportLibraries-v0.5.0%2B0/CompilerSupportLibraries.v0.5.0.\${target}-libgfortran3.tar.gz" | tar xzf - -C \${prefix}
```

Comment:
I'm a bit in a rush now and I wanted to get this PR out for testing, but this needs to use the automatic mechanism to install dependencies, instead of manually doing this non-cache-friendly download inside the build.